cromwell_tools/cromwell_api.py

```diff
@@ -0,0 +1,439 @@
+"""
+TODO: add some module docs
+"""
+
+import time
+import tempfile
+from subprocess import PIPE, Popen
+import os
+import json
+import logging
+import requests
+from datetime import datetime, timedelta
+from tenacity import retry, stop_after_delay, wait_exponential
+
+from cromwell_tools.utilities import validate_cromwell_label, _localize_file
+
+
+logger = logging.getLogger(__name__)
+
+
+class WorkflowFailedException(Exception):
+    pass
+
+
+class WorkflowUnknownException(Exception):
+    pass
+
+
+# TODO: use functools partial for get, post (set the authenticate commands)
+class CromwellAPI(object):
+    """Contains a set of classmethods that implement interfaces to cromwell REST API endpoints."""
+
+    # TODO: move the endpoints definitions to the corresponding functions after refactoring the unit tests and mocks
+    _abort_endpoint = '/api/workflows/v1/{uuid}/abort'
```

Comment:
I meant to do this when I was working on the PR, but I prefer to move the endpoints into the corresponding functions. Also, what do you think about defining the other constants (`_failed_statuses`, `_cromwell_exclusive_query_keys`,  `_cromwell_inclusive_query_keys`, and `_cromwell_query_keys `) outside of this class? Even though there is an `_` indicating it is "private", it is still possible to change these values.

Reply:
I agree with both of the ideas! Do you think we need to do that in this PR or we could do it in a follow-up PR?

Reply:
Lets move these constants out of the class for now, and then we can move the endpoints into the corresponding functions later!